main.js

```diff
@@ -160,4 +160,175 @@ const inventory = [
+const listOfTVBrands = document.getElementById('listTvBrands');
+const arrayOfTVBrands = inventory.map((tv) => {
+  return tv.brand;
+})
+
+// De Set methode zorgt ervoor dat er een array overblijft met unieke waarden.
+const uniqueArrayofTVBrands = [...new Set(arrayOfTVBrands)];
+
+for (let i = 0; i < uniqueArrayofTVBrands.length; i++) {
+  listOfTVBrands.innerHTML += `
+  <li>${ uniqueArrayofTVBrands[i] }</li>
+`
+}
+
+// Opdracht 3b: Schrijf de code uit 3a om naar een functie die een array met tv-objecten verwacht. Het is handig om onze scripts als functies op te zetten, zodat we ze gemakkelijk kunnen hergebruiken. Tip: vergeet deze functie -declaratie niet aan te roepen!
+function createListOfTVBrands(inventory) {
+  const arrayOfTVBrands = inventory.map((tv) => {
+    return tv.brand;
+  });
+  const uniqueArrayofTVBrands = [...new Set(arrayOfTVBrands)];
+  return uniqueArrayofTVBrands;
+}
+
+console.log(createListOfTVBrands(inventory));
+
+
+/*----------------------
+    Opdracht 4 - Functies
+----------------------*/
+// Maak deze gehele opdracht eerst alsof je het voor één tv doet. We gaan één tv weergeven in het volgende format:
+//     Philips 43PUS6504/12 - 4K TV
+//     €379,-
+//     43 inch (109 cm) | 50 inch (127 cm) | 58 inch (147 cm)
+
+// Opdracht 4a: Maak een herbruikbare functie die een string genereert voor de naam van één tv en deze teruggeeft in het format [merk] [type] - [naam] zoals Philips 43PUS6504/12 - 4K TV of NIKKEI NH3216SMART - HD smart TV.
+function getTVName(tv) {
+  return `${tv.brand} ${tv.type} - ${tv.name}`;
+}
+
+// Opdracht 4b: Maak een herbruikbare functie die de prijs van één tv als parameter verwacht (zoals 379 of 159) teruggeeft in het format €379,- of €159,-.
+function getPrice(tv) {
+  return `€${tv.price},-`;
+}
+
+// Opdracht 4c: Maak een herbruikbare functie die een string genereert voor alle beschikbare schermgroottes van één tv. De functie geeft dit terug in het format [schermgrootte] inches ([schermgrootte omgerekend]cm) | [schermgrootte] inches ([schermgrootte omgerekend]cm) etc. Als een tv maar één schermgrootte heeft ([32]) wordt de output 32 inch (81 cm). Wanneer een tv vier schermgroottes heeft ([43, 50, 55, 58]) wordt de output 43 inch (109 cm) | 50 inch (127 cm) | 58 inch (147 cm). Let op: om één string te genereren uit een array van schermgroottes zul je een for-loop voor moeten gebruiken.
+function getScreenSize(tv) {
+  const screenSizes = tv.availableSizes;
+  const screenSizesArray = [];
+  for (let i = 0; i < screenSizes.length; i++) {
+    const screenSizeInch = screenSizes[i];
+    const screenSizeCM = screenSizes[i] * 2.54;
+    screenSizesArray.push(`${screenSizeInch} inch (${screenSizeCM} cm)`);
+  }
+  if (screenSizesArray.length === 1) {
+    return screenSizesArray[0];
+  } else {
+    return screenSizesArray.join(" | ")
+  }
```

Comment:
Als het goed is heeft dit hetzelfde resultaat
```suggestion
  return screenSizesArray.join(" | ")
```